draft-ppm-protocol.md

```diff
@@ -672,51 +686,70 @@ continuation of a previous sub-batch?]
 ### Aggregate Request
 
 The AggregateReq request is used by the leader to send a set of reports to the
-helper. These reports MUST all be associated with the same PPM task. [[OPEN
-ISSUE: And the same batch, right?]]
+helper. These reports MUST all be associated with the same PPM task. The reports
+MAY span multiple batches, but MUST be transmitted in the order specified in
+{{anti-replay}}.
 
-For each aggregator endpoint `[aggregator]` in `AggregateReq.task_id`'s
-parameters except its own, the leader sends a POST request to
-`[aggregator]/aggregate` with the following message:
+For the second aggregator endpoint `[aggregator]` in `AggregateReq.task_id`'s
+parameters, the leader sends a POST request to `[aggregator]/aggregate` with the
+following message:
 
 ~~~
 struct {
+  AggregateReqType type;
   TaskID task_id;
   opaque helper_state<0..2^16>;
+  select (type) {
+    case start: opaque output_param<0..2^16-1>;
+    case next:  Empty;
+  }
   AggregateSubReq seq<1..2^24-1>;
 } AggregateReq;
+
+enum { start(0), next(1), } AggregateReqType;
 ~~~
 
-The structure contains the PPM task, an opaque *helper state* string, and a
-sequence of *sub-requests*, each corresponding to a unique client report.
-Sub-requests are structured as follows:
+The structure contains the PPM task, an opaque *helper state* string
+(`helper_state`), and a sequence of *sub-requests* (`seq`), each corresponding
+to a unique client report. The first aggregate request also carries the output
+parameter (`output_param`) used for VDAF evaluation. Sub-requests are
+structured as follows:
 
 ~~~
 struct {
-  Time time;                       // Equal to Report.time.
-  uint64 nonce;                    // Equal to Report.nonce.
-  Extension extensions<4..2^16-1>; // Equal to Report.extensions.
-  EncryptedInputShare helper_share;
-  select (protocol) { // Protocol for the PPM task
-    case prio: PrioAggregateSubReq;
-    case hits: HitsAggregateSubReq;
+  Time time;                       // Report.time
+  uint64 nonce;                    // Report.nonce
+  Extension extensions<4..2^16-1>; // Report.extensions
+  opaque verify_message<1..2^16-1>;
+  select (AggregateReq.type) {
+    case start: EncryptedInputShare helper_share;
+    case next:  Empty;
   }
 } AggregateSubReq;
+
 ~~~
 
-The `time`, `nonce`, and `extensions` fields have the same value as those in the
-report uploaded by the client. Similarly, the `helper_share` field is the
-`EncryptedInputShare` from the `Report` whose index in
-`Report.encrypted_input_shares` is equal to the index of `[aggregator]` in the
-task's aggregator endpoints. [OPEN ISSUE: We usually only need to send this in
-the first aggregate request. Shall we exclude it in subsequent requests
-somehow?] The remainder of the structure is dedicated to the protocol-specific
-request parameters.
+Each sub-request for a given report includes the report's timestamp, nonce, and
```

Comment:
I don't understand "each sub-request for a given report". Is there some other structure?

"Each subrequest corresponds to a single report"

Reply:
Good catch. Should be "Each sub-request includes the report's ..."